File: thugtools/src/main/java/com/siddhantkushwaha/thugtools/GsonUtil.java
package com.siddhantkushwaha.thugtools;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonElement;

import java.lang.reflect.Type;

public class GsonUtil {
    private static final Gson gson = new GsonBuilder().disableHtmlEscaping().create();

    public static String toGson(Object object) {
        return gson.toJson(object);
    }

    public static <T> T fromGson(String json, Class<T> type) {
        return gson.fromJson(json, type);
    }

    public static <T> T fromGson(JsonElement json, Class<T> type) {
        return gson.fromJson(json, type);
    }

    public static <T> T fromGson(String json, Type type) {
        return gson.fromJson(json, type);
    }

    public static <T> T fromGson(JsonElement json, Type type) {
        return gson.fromJson(json, type);
    }
}

File: thugtools/src/main/java/com/siddhantkushwaha/thugtools/GsonUtil.kt
package com.siddhantkushwaha.thugtools

import com.google.gson.GsonBuilder
import com.google.gson.JsonElement
import java.lang.reflect.Type


object GsonUtil {
    private val gson = GsonBuilder().disableHtmlEscaping().create()

    fun toGson(obj: Any?): String {
        return gson.toJson(obj)
    }

    fun <T> fromGson(json: String?, type: Class<T>?): T {
        return gson.fromJson(json, type)
    }

    fun <T> fromGson(json: JsonElement?, type: Class<T>?): T {
        return gson.fromJson(json, type)
    }

    fun <T> fromGson(json: String?, type: Type?): T {
        return gson.fromJson(json, type)
    }

    fun <T> fromGson(json: JsonElement?, type: Type?): T {
        return gson.fromJson(json, type)
    }
}